modules/sdk-coin-near/src/near.ts

```diff
@@ -1077,4 +1093,207 @@ export class Near extends BaseCoin {
       }
     }
   }
+
+  private validateTokenEnablementTransaction(
+    transaction: Transaction,
+    explainedTx: TransactionExplanation,
+    txParams: TransactionParams
+  ): void {
+    this.validateTxType(txParams, explainedTx);
+    this.validateSigner(transaction);
+    this.validateRawReceiver(transaction, txParams);
+    this.validatePublicKey(transaction);
+    this.validateRawActions(transaction, txParams);
+    this.validateBeneficiary(explainedTx, txParams);
+    this.validateTokenOutput(explainedTx, txParams);
```

Comment:
This one is more like a nit for optimization:
The idea is to remove the internal transaction.toJson() calls that I'll assume could be heavy ones and just pass the parsed tx data down
```suggestion
    const transactionData = transaction.toJson();
    this.validateTxType(txParams, explainedTx);
    this.validateSigner(transactionData);
    this.validateRawReceiver(transactionData, txParams);
    this.validatePublicKey(transactionData);
    this.validateRawActions(transactionData, txParams);
    this.validateBeneficiary(explainedTx, txParams);
    this.validateTokenOutput(explainedTx, txParams);
```